File: aidlc-table-order/packages/shared/src/types/recommendation.ts
import type { Menu } from './menu';

export interface RecommendationRequest {
  menu_ids: string[];
}

export interface RecommendedMenu {
  id: string;
  name: string;
  price: number;
  image_url: string | null;
}

export interface RecommendationResponse {
  recommendations: RecommendedMenu[];
}

File: aidlc-table-order/backend/app/dto/recommendation.py
"""Recommendation DTOs — request/response models for C6: Recommendation."""

from pydantic import BaseModel


class RecommendationRequest(BaseModel):
    menu_ids: list[str]


class RecommendedMenu(BaseModel):
    id: str
    name: str
    price: int
    image_url: str | None = None


class RecommendationResponse(BaseModel):
    recommendations: list[RecommendedMenu]
